tests/test_basis.h

```diff
@@ -31,10 +31,258 @@
 #ifndef TEST_BASIS_H
 #define TEST_BASIS_H
 
-#include "core/os/main_loop.h"
+#include "core/math/random_number_generator.h"
+#include "core/os/os.h"
+#include "core/ustring.h"
+
+#include "tests/test_macros.h"
 
 namespace TestBasis {
-MainLoop *test();
+
+enum RotOrder {
+	EulerXYZ,
+	EulerXZY,
+	EulerYZX,
+	EulerYXZ,
+	EulerZXY,
+	EulerZYX
```

Comment:
Only YXZ is actually used, we can remove all the others. To be more precise, they're not exposed, and their only usage in the engine is in the test cases.

Reply:
Perhaps we expose them! I remember this is a prerequisite for maya fbx support.

Reply:
If something is mathematically possible to calculate, I believe the tests should cover those as well, to provide more robustness, we may get even more stable `YXZ` computation as a byproduct.

Reply:
My decision is keep these tests because they're mathematically possible.